File: Data Structures & Algorithms/is-subsequence/submission-1.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        if (s.length() == 0) {
            return true;
        }

        int curr = 0;
        for (int i = 0; i < t.length(); i++) {
            if (t.charAt(i) == s.charAt(curr)) {
                curr++;
                if (curr == s.length()) {
                    return true;
                }
            }
        }

        return false;
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if root is None:
            return []

        queue = []
        queue.append(root)
        curr_iteration_num = len(queue)
        curr_idx = 0
        res = []

        while bool(queue):
            curr = queue.pop(0)
            if curr.left != None:
                queue.append(curr.left)
            if curr.right != None:
                queue.append(curr.right)

            curr_iteration_num -= 1

            if len(res) == curr_idx:
                res.append([curr.val])
            else:
                res[curr_idx].append(curr.val)

            if curr_iteration_num == 0:
                curr_iteration_num = len(queue)
                curr_idx += 1

        return res